File: src/dataset/typescript/run-length-encoding_exercism-samajammin/gemini/GeminiFlash_run-length-encoding_v2.ts
export default class RunLengthEncoding {
  static encode(str: string): string {
    if (!str) {
      return "";
    }

    let res = '';
    let count = 1;

    for (let i = 0; i < str.length; i++) {
      if (i + 1 < str.length && str[i] === str[i + 1]) {
        count++;
      } else {
        res += (count > 1 ? count : "") + str[i];
        count = 1;
      }
    }

    return res;
  }

  static decode(str: string): string {
    let res = '';
    let num = '';

    for (let i = 0; i < str.length; i++) {
      const char = str[i];
      if (/[0-9]/.test(char)) {
        num += char;
      } else {
        const repeatCount = num ? parseInt(num, 10) : 1;
        res += char.repeat(repeatCount);
        num = '';
      }
    }

    return res;
  }
}

File: src/dataset/python/run-length-encoding_exercism-fruit-in/gemini/GeminiFlash_run_length_encoding_v2.py
from itertools import groupby
import re


def decode(string):
    decompressed = []
    for count, char in re.findall(r'(\d*)(\D)', string):
        num = int(count) if count else 1
        decompressed.append(char * num)
    return ''.join(decompressed)


def encode(string):
    compressed = []
    for char, group in groupby(string):
        count = sum(1 for _ in group)
        compressed.append(char if count == 1 else str(count) + char)
    return ''.join(compressed)